LPS-89103 data record collection permissions

rbohl opened the pull request:
Ready for review: https://issues.liferay.com/browse/LPS-89103, cc @marcelabc

rbohl committed:
LPS-89103, draft for defining data record collection permissions

rbohl committed:
LPS-89103 light editing

liferay-continuous-integration commented:
To conserve resources, the PR Tester does not automatically run for every pull.

If your code changes were already tested in another pull, reference that pull in this pull so the test results can be analyzed.

If your pull was never tested, comment &quot;ci:test&quot; to run the PR Tester for this pull.

marcelabc commented on developer/frameworks/articles/data-engine/02-data-collection-api/00-intro.markdown:
```diff
@@ -0,0 +1,71 @@
+
+Data Record Collections (`DEDataRecordCollection`s) are used to collect
+`DEDataRecord`s. A Data Definition defines the fields present in the Data
+Collection, while the Data Record Collection forms a boundary, collecting only
```
A Data Definition defines the fields present in the Data Record, ...

marcelabc commented on developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown:
```diff
@@ -0,0 +1,146 @@
+- `allowDelete()` to grant deletion permission
+- `allowUpdate()` to grant update permission
+- `allowView()` to grant view permission
+- `allowAddDataRecord()` to grant add permissions, to add `DEDataRecord`
```
remove the comma, please :)

marcelabc commented:
I just reviewed and it is fine for me 👍

rbohl commented:
Thanks! I'm about to start writing up the other data record collection operations. Is it okay to just push the commits into my branch and update this PR?

@marcelabc

rbohl committed:
LPS-87772, deleting data record collections

rbohl committed:
LPS-89167 DEDataRecordCollectionGetRequest

rbohl committed:
LPS-90091 build a DEDataRecordCollectionGetRecordRequest

rbohl committed:
LPS-87620 (editing), LPS-89258 (adding), and LPS-89133 (deleting) data
records

rbohl committed:
LPS-89258, creating and saving data records

rbohl committed:
LPS-90790, getting a list of data collections

rbohl commented:
@marcelabc This is ready for review. I'll put the tickets pertaining to the new information in review now.

marcelabc commented on developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown:
```diff
@@ -152,6 +152,14 @@ To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteReques
 
+## Deleting a Data Record
+
+To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteRequest` to
```
"To delete a **DEDataRecord**, pass a **DEDataRecordCollectionDeleteRecordRequest** to ..

marcelabc commented on developer/frameworks/articles/data-engine/02-data-collection-api/01-data-record-collection-operations.markdown:
```diff
@@ -152,6 +152,14 @@ To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteReques
+To delete a `DEDataRecordCollection`, pass a `DEDataRecordCollectionDeleteRequest` to
+`DEDataRecordCollectionService.execute()`.
+
+Required parameter: `long deDataRecordCollectionId`
```
Required parameter: long **deDataRecordId**

marcelabc commented:
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

marcelabc committed:
LPS-89103 Correction on the definition of the "Data Definition

marcelabc committed:
LPS-89103 Remove the comma

marcelabc committed:
LPS-89130 Correction on the delete definition (Data Record)

marcelabc committed:
LPS-89133 [Delete Data Record Collection] Add more details in how construct the request

marcelabc committed:
LPS-89167 [Get Data Record Collection] Add more details in how construct the request

marcelabc committed:
LPS-89258 [Save/Update Data Record] Add more details in how construct the request

marcelabc committed:
LPS-90091 Fix

rbohl committed:
LPS-91158, listing data records

rbohl committed:
LPS-90942 searching for data collections

LPS-90942, incomplete--searching for data record collections

LPS-90942 searching for data collections

rbohl commented:
@gabrielwas @marcelabc I pushed up sections on building requests for:

listing data records -- https://github.com/gabrielwas/liferay-docs/pull/4/commits/2c1b6ac0a6eb49129aaf541f2c2b0b3752b94bc3

searching for data collections -- https://github.com/gabrielwas/liferay-docs/pull/4/commits/155367757ce5a933d5495b11214a1d1ceaf9ea16
